exporter/datadogexporter/agent_components.go

```diff
@@ -0,0 +1,178 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadogexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter"
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	coreconfig "github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder"
+	"github.com/DataDog/datadog-agent/comp/forwarder/orchestrator/orchestratorinterface"
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp/components/exporter/serializerexporter"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes"
+	otlpmetrics "github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/metrics"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter"
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxevent"
+	"go.uber.org/zap"
+)
+
+func newLogComponent(set component.TelemetrySettings) (log.Component, error) {
+	zlog := &zaplogger{
+		logger: set.Logger,
+	}
+	return zlog, nil
+}
+
+func newSerializer(set component.TelemetrySettings, cfg *Config) (*serializer.Serializer, error) {
+	var f defaultforwarder.Component
+	var c coreconfig.Component
+	var s *serializer.Serializer
+	yamldata := fmt.Sprintf(`logs_enabled: true
+log_level: %s
+site: %s
+api_key: %s
+apm_config:
+  enabled: true
+  apm_non_local_traffic: true
+forwarder_timeout: 10`, set.Logger.Level().String(), cfg.API.Site, cfg.API.Key)
```

Comment:
I believe API key will be printed as `[REDACTED]` since this an opaque string, you probably have to cast to string